crates/astria-proto/proto/astria/sequencer_relayer/v1/block.proto

```diff
@@ -0,0 +1,20 @@
+syntax = "proto3";
+
+package astria.sequencer_relayer.v1;
+
+import "tendermint/types/types.proto";
+import "astria/sequencer_relayer/v1/data.proto";
+
+// SequencerBlock
+message SequencerBlock {
+    // helper type - these should get parsed into a map from namespace to a vector of IndexedTransactions
+    message NamespacedIndexedTransactions {
+        bytes namespace = 1;
+        repeated IndexedTransaction txs = 2;
+    }
+
+    bytes block_hash = 1;
+    tendermint.types.Header header = 2;
+    repeated IndexedTransaction sequencer_txs = 3;
+    repeated NamespacedIndexedTransactions rollup_txs = 4;
```

Comment:
So these deeply nested types are a bit tricky form a performance point of view:

1. protobuf does not contain length information in the wire-format for `repeated` messages (which includes `map`, because these are just `repeated` `(key,value)` tuples);
2. protobuf does not guarantee that `repeated` messages are contiguous on the wire (the first element might be in the beginning, the last element at the end of the bytestream);

This means there is no way to pre-allocate sufficiently sized `Vec`s. And so you will end up with a ton of small allocations in the child-vecs which cannot get amortized.

As an alternative route I would suggest making use of a technique frequently encountered in game dev by inlining all the fields, i.e.:

```protobuf3
message NamespacedIndexedTransaction {
    bytes namespace = 1;
    // Maybe with a clearer name here what the index exactly refers to.
    uin64 index = 2;
    bytes transaction = 3;
}
```

Reply:
see issue #31